go-controller/pkg/network-attach-def-controller/network_attach_def_controller.go

```diff
@@ -247,6 +258,11 @@ func (nadController *NetAttachDefinitionController) syncNAD(key string, nad *net
 		} else {
 			nadController.networkManager.EnsureNetwork(oldNetwork)
 		}
+		// if oldNetwork is primary, and we are not going to update the active network, delete it
+		// as long as no other NADs reference it
```

Comment:
Although it seems all of the code is protected against receiving NAD events out of order somehow, since we can have multiple primary network per namespace in the map, so also probably worth commenting this detail.

Reply:
I think my comment is outdated and should just say:
// if oldNetwork is primary, then delete it as long as no other NADs reference it

I'm not sure what your comment is describing. There can be multiple primary NADs per namespace (although there shouldn't be), and we have to handle this type of scenario. We only protect against multiple primary NADs in the case where a user only uses UDN CRD.

Reply:
The specifically confusing part here is that the `no other NADs reference it` comment makes it look like the `unsetActiveNetworkForNamespace` call belongs in the `if len(oldNetwork.GetNADs()) == 0` block above, 

I know that's not true but the comment doesn't help.

Reply:
I'll just remove the comment then? There are comments in the unset function itself.